src/extra/widgets/menu/lv_menu.h

```diff
@@ -0,0 +1,238 @@
+
+#if LV_USE_MENU
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+enum {
+    LV_MENU_MODE_HEADER_TOP_FIXED, /* Header is positioned at the top */
+    LV_MENU_MODE_HEADER_TOP_UNFIXED, /* Header is positioned at the top and can be scrolled out of view*/
+    LV_MENU_MODE_HEADER_BOTTOM_FIXED /* Header is positioned at the bottom */
+};
+typedef uint8_t lv_menu_mode_header_t;
+
+enum {
+    LV_MENU_MODE_ROOT_BACK_BTN_DISABLED,
+    LV_MENU_MODE_ROOT_BACK_BTN_ENABLED
+};
+typedef uint8_t lv_menu_mode_root_back_btn_t;
+
+enum {
+    LV_MENU_MODE_SIDEBAR_DISABLED,
+    LV_MENU_MODE_SIDEBAR_ENABLED
+};
+typedef uint8_t lv_menu_mode_sidebar_t;
+
+typedef struct lv_menu_load_page_event_data_t {
+    lv_obj_t * menu;
+    lv_obj_t * page;
+} lv_menu_load_page_event_data_t;
+
+typedef struct {
+    lv_obj_t * page;
+} lv_menu_history_t;
+
+typedef struct {
+    lv_obj_t obj;
+    lv_obj_t * storage; /* a pointer to obj that is the parent of all pages not displayed */
+    lv_obj_t * main;
+    lv_obj_t * main_page;
+    lv_obj_t * main_header;
+    lv_obj_t * main_header_back_btn; /* a pointer to obj that on click triggers back btn event handler, can be same as 'main_header' */
+    lv_obj_t * sidebar;
+    lv_obj_t * sidebar_page;
+    lv_obj_t * sidebar_header;
+    lv_obj_t * sidebar_header_back_btn; /* a pointer to obj that on click triggers back btn event handler, can be same as 'sidebar_header' */
+    lv_obj_t * selected_tab;
+    lv_ll_t history_ll;
+    uint8_t cur_depth;
+    uint8_t prev_depth;
+    uint8_t sidebar_generated : 1;
+    lv_menu_mode_header_t mode_header : 2;
+    lv_menu_mode_root_back_btn_t mode_root_back_btn : 1;
+    lv_menu_mode_sidebar_t mode_sidebar : 1;
+} lv_menu_t;
+
+extern const lv_obj_class_t lv_menu_class;
+extern const lv_obj_class_t lv_menu_page_class;
+extern const lv_obj_class_t lv_menu_cont_class;
+extern const lv_obj_class_t lv_menu_section_class;
+extern const lv_obj_class_t lv_menu_separator_class;
+extern const lv_obj_class_t lv_menu_sidebar_cont_class;
+extern const lv_obj_class_t lv_menu_main_cont_class;
+extern const lv_obj_class_t lv_menu_sidebar_header_cont_class;
+extern const lv_obj_class_t lv_menu_main_header_cont_class;
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**
+ * Create a menu object
+ * @param parent pointer to an object, it will be the parent of the new menu
+ * @return pointer to the created menu
+ */
+lv_obj_t * lv_menu_create(lv_obj_t * parent);
+
+/**
+ * Create a menu page object
+ * @param parent pointer to menu object
+ * @return pointer to the created menu page
+ */
+lv_obj_t * lv_menu_page_create(lv_obj_t * parent);
+
+/**
+ * Create a menu cont object
+ * @param parent pointer to an object, it will be the parent of the new menu cont object
+ * @return pointer to the created menu cont
+ */
+lv_obj_t * lv_menu_cont_create(lv_obj_t * parent);
```

Comment:
You are correct that it is not absolutely required to use these. Layout can even be a grid pattern. I added in because it makes for theming as in using lv_theme_default easier. The in the default theme this container has styles for checked and clicked state.